### PR TITLE
Don't html-ize high rank admins in Discord.

### LINKED ISSUE #25115: The new tickets thing needs HTML tags stripped
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

N/A

### Issue Description

image

This can likely be fixed using the same code thats in runechat to strip HTML tags out.

### What did you expect to happen?

No tags

### What happened instead?

Tags

### Why is this bad/What are the consequences?

Looks naff

### Steps to reproduce the issue.

Make a ticket reply

### When did the problem start happening?

Since #24719

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
High-rank admin messages to Discord were getting HTML included. They shouldn't.
Removed `strip_html_properly`, which made a few other things need minor adjustments.
Fixes log_debug so nobody else gets tricked by it like I did.

Fixed #25115

## Why It's Good For The Game
HTML in discord bad.

## Testing
Fixed log_debug, logged messages sent to discord

## Changelog
:cl:
fix: Cleaned up escaping in a few more places. This likely fixes HTML elements appearing in whisper text and overheard text while asleep.
/:cl:

### CODE REVIEW

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -346,35 +346,21 @@
...
-				input = copytext(input, 1, opentag) + copytext(input, (closetag + 1))
-		else if(closetag || opentag)
-			if(opentag)
-				input = copytext(input, 1, opentag)
-			else
-				input = copytext(input, (closetag + 1))
-		else
-			break
+		return ""
+	var/global/regex/tags = regex("<\[^>]*>", "g")
```
Comment (Contrabang) [inline code/__HELPERS/text.dm:354]: does this need to be global? or can it be static

Comment (FunnyMan3595) [reply code/__HELPERS/text.dm:354]: So, funny thing, global and static are **identical** in DM. The reference only lists global, static is just an alias for it.

Yes, this is exactly as dumb as it sounds.

Comment (GDNgit) [reply code/__HELPERS/text.dm:354]: I can show you something even dumber, static vars within functions get regenerated a lot for almost no reason. So you need to see if the var is set first then regenerate it if it's not.

Comment (GDNgit) [inline code/__HELPERS/text.dm:354]:
```suggestion
	var/static/regex/tags = regex("<\[^>]*>", "g")
	if(!tags)
		tags = regex("<\[^>]*>", "g")
```
static is generally preferred for this due to readability reasons.

**Review by Burzah — APPROVED** [review]

File: code/modules/admin/verbs/adminpm.dm [hunk 2]
```diff
@@ -104,12 +104,8 @@
...
 		return
 
-	//clean the message if it's not sent by a high-rank admin
-	if(!check_rights(R_SERVER|R_DEBUG,0))
-		msg = sanitize_simple(copytext_char(msg, 1, MAX_MESSAGE_LEN))
-		if(!msg)
-			return
-	else
+	// Let high-rank admins use advanced pencode.
+	if(check_rights(R_SERVER|R_DEBUG,0))
```
Comment (DGamerL) [inline code/modules/admin/verbs/adminpm.dm:108]:
```suggestion
	if(check_rights(R_SERVER|R_DEBUG, 0))
```

File: code/modules/mob/hear_say.dm [hunk 3]
```diff
@@ -174,16 +174,16 @@
...
-		message = strip_html_properly(message)
+		message = html_decode(strip_html_tags(message))
 		var/list/punctuation = list(",", "!", ".", ";", "?")
 		var/list/messages = splittext(message, " ")
 		if(messages.len > 0)
 			var/R = rand(1, messages.len)
 			var/heardword = messages[R]
 			if(copytext(heardword,1, 1) in punctuation)
-				heardword = copytext(heardword,2)
+				heardword = html_encode(copytext(heardword,2))
```
Comment (DGamerL) [inline code/modules/mob/hear_say.dm:184]:
```suggestion
				heardword = html_encode(copytext(heardword, 2))
```

File: code/modules/mob/hear_say.dm [hunk 4]
```diff
@@ -174,16 +174,16 @@
...
 			if(copytext(heardword,-1) in punctuation)
-				heardword = copytext(heardword,1,length(heardword))
+				heardword = html_encode(copytext(heardword,1,length(heardword)))
```
Comment (DGamerL) [inline code/modules/mob/hear_say.dm:186]:
```suggestion
				heardword = html_encode(copytext(heardword, 1, length(heardword)))
```

**Review by Contrabang — APPROVED** [review]
I can't spot anything wrong with it, but I'd like a second approval on it

**Review by DGamerL — APPROVED** [review]

**Pull request merged**